### PR TITLE
Pin the version of tools used in CI

### PR DESCRIPTION
**Pull request by alalazo**
Extracted from #37041 

Modifications:
- [x] Pin the version of tools we use in CI for style validation
- [x] Have dependabot manage dependency update

### CODE REVIEW

**Comment by haampie** [comment]
Can you instead fix `major.minor.*` (or even `major.*`)

**Comment by alalazo** [comment]
What would be the benefit of not pinning exactly, if dependabot takes care of the update? I did this out of occasional complaints where a bump in versions (even minor) breaks develop. In this way we'll see the breakage in a PR.

File: .github/workflows/style/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+black==23.1.0
+clingo==5.6.2
+flake8==6.1.0
+isort==5.12.0
+mypy==1.5.0
+types-six==1.16.21.9
```
Comment (adamjstewart) [inline .github/workflows/style/requirements.txt:6]: Are we still using six??

**Review by adamjstewart — APPROVED** [review]
I prefer exact pins

File: .github/workflows/style/requirements.txt (hunk 1 above)
Comment (alalazo) [reply .github/workflows/style/requirements.txt:6]: Some of our style dependencies was, not sure which one

**Pull request merged**